### ISSUE TITLE
`useCreate` requires an `id` property

### ISSUE THREAD

**Issue by andrico1234** [marmelab/react-admin#7459]
**What you were expecting:**
For a call to `useCreate` like this:

```
  const [createUser] = useCreate(
    "users",
    {
      data: {
        username,
      },
    },
)
```

I expect typescript to not throw an errors,

after the latest rc-0 update I'm getting the following compilation error: 

`Property 'id' is missing in type '{ clientId: any; bookingId: any; userId: any; }' but required in type 'RaRecord'.ts(2741)`

I can only guess that the latest change [here](https://github.com/marmelab/react-admin/pull/7431) caused the `id` property to be required. 

This is likely a mistake, since most records won't know ahead of time their ID, since it's a value created once the record's been inserted into the database.

**Environment**

* React-admin version:
* Last version that did not exhibit the issue (if applicable):
* React version:
* Browser:
* Stack trace (in case of a JS error):

**Comment by WiXSL** [comment]
Reproduced.
Thanks!

**Comment by fzaninotto** [comment]
Fixed by #7460
